app/app/api/deps.py

```diff
@@ -26,6 +26,31 @@ def get_current_user(db: Session = Depends(get_db), token: str = Depends(reusabl
     try:
         payload = jwt.decode(token, settings.SECRET_KEY, algorithms=[security.ALGORITHM])
         token_data = schemas.TokenPayload(**payload)
+        if token_data.type != 'access':
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Invalid token",
+            )
+    except (jwt.JWTError, ValidationError):
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Could not validate credentials",
+        )
+    user = crud.user.get(db, id=token_data.sub)
+    if not user:
+        raise HTTPException(status_code=404, detail="User not found")
+    return user
+
+
+def get_current_user_refresh(db: Session = Depends(get_db), token: str = Depends(reusable_oauth2)) -> models.User:
```

Comment:
what is the purpose of this new function? The code seems to be the same, except for it allowing tokens of type `refresh` instead of access.

Reply:
Yep, basically just that
Could probably rewrite it to work like our admin permission check functions.